libraries/bot-azure/src/main/java/com/microsoft/bot/azure/blobs/BlobsTranscriptStore.java

```diff
@@ -336,33 +345,38 @@ public BlobsTranscriptStore(String dataConnectionString, String containerName) {
                     String prefix = String.format("%s/%s/",
                         sanitizeKey(activity.getChannelId()), sanitizeKey(activity.getConversation().getId()));
                     Iterable<PagedResponse<BlobItem>> resultSegment = containerClient
-                        .listBlobsByHierarchy(prefix).iterableByPage(token);
+                        .listBlobsByHierarchy("/",
+                            this.getOptionsWithMetadata(prefix), null).iterableByPage(token);
                     token = null;
-
                     for (PagedResponse<BlobItem> blobPage: resultSegment) {
                         for (BlobItem blobItem: blobPage.getValue()) {
                             if (blobItem.getMetadata().get("Id").equals(activity.getId())) {
                                 BlobClient blobClient = containerClient.getBlobClient(blobItem.getName());
-                                this.getActivityFromBlobClient(blobClient).thenApply(blobActivity -> CompletableFuture
-                                    .completedFuture(new Pair<Activity, BlobClient>(blobActivity, blobClient)));
+                                Activity blobActivity = this.getActivityFromBlobClient(blobClient).join();
```

Comment:
I added that `.join()` because I had issues with the `while()`, I needed to return something in order to finish the loop. Anyway, now I tried using a `thenApply` and it's working good

Reply:
Good to know